taxcalc/functions.py

```diff
@@ -826,14 +849,28 @@ def AMTI(c60000, _exact, e60290, _posagi, e07300, x60260, c24517,
     c60240 = (1 - ID_StateLocalTax_HC) * c18300 + x60240
     c60220 = c20800 + x60220
     c60130 = c21040 + x60130
+    # imputation for x62730
+    if e62730 != 0 and e24515 > 0:
+        x62730 = e62730 - e24515
     c62730 = e24515 + x62730
 
     _amtded = c60200 + c60220 + c60240
+    _prefded = c60200 + c60220 + c60240
+    _prefnot = c21060 - c21040 - _prefded
+    _totded = _prefded + _prefnot
+    _useded = min(_totded, max(0, c00100 - c04600))
+    c04500 = c00100 - max(_useded, _standard + e37717)
+    if FDED == 1:
+        c60000 = c00100 - _useded
+    else:
```

Comment:
Dan mentioned that he was not sure about the AMT part, so I'll skip this part at this moment.